test/Core.Test/Utilities/RequireFeatureAttributeTests.cs

```diff
@@ -23,8 +22,14 @@ public void Throws_When_Feature_Disabled()
         // Arrange
         var rfa = new RequireFeatureAttribute(_testFeature);
 
-        // Act & Assert
-        Assert.Throws<FeatureUnavailableException>(() => rfa.OnActionExecuting(GetContext(enabled: false)));
+        // Act
+        var context = GetContext(enabled: false);
+        rfa.OnActionExecuting(context);
+
+        // Assert
+        Assert.IsType<NotFoundObjectResult>(context.Result);
+        var result = (NotFoundObjectResult)context.Result;
+        Assert.Contains("This feature is unavailable", (string)result.Value);
```

Comment:
❓ Do we need to assert error messages in tests?

Reply:
It's our current practice. In this case (for example), it ensures that the `NotFoundObjectResult` is the one I expect and not for some other reason (a false negative).